### PR TITLE
Use new Attributes hash per JSON-API changes

### PR DESCRIPTION
**Pull request by lgebhardt**

### CODE REVIEW

File: test/controllers/controller_test.rb [hunk 1]
```diff
@@ -1096,6 +1143,31 @@ def test_update_missing_type
...
+    set_content_type_header!
+    javascript = Section.find_by(name: 'javascript')
+
+    put :update,
+        {
+          id: 3,
+          data: {
+            id: '3',
+            type: 'posts',
+            body: 'asdfg',
```
Comment (dgeb) [inline test/controllers/controller_test.rb:1156]: isn't `body` an attribute?

Comment (lgebhardt) [inline test/controllers/controller_test.rb:1156]: Yes, that's what this test is supposed to catch.

File: test/controllers/controller_test.rb [hunk 2]
```diff
@@ -1096,6 +1143,31 @@ def test_update_missing_type
     assert_match /The required parameter, type, is missing./, response.body
   end
 
+  def test_update_unknown_key
+    set_content_type_header!
+    javascript = Section.find_by(name: 'javascript')
+
+    put :update,
+        {
+          id: 3,
```
Comment (dgeb) [inline test/controllers/controller_test.rb:1152]: extra `id`?

File: test/controllers/controller_test.rb (hunk 1 above)
Comment (dgeb) [inline test/controllers/controller_test.rb:1156]: Ah, I see!

File: test/controllers/controller_test.rb (hunk 2 above)
Comment (lgebhardt) [inline test/controllers/controller_test.rb:1152]: The top level `id` is provided by the route and wouldn't be part of the payload.

**Comment by dgeb** [comment]
looks good @lgebhardt :+1:

**Pull request merged**